contracts/truefi2/LoanToken2.sol

```diff
@@ -375,16 +371,17 @@ contract LoanToken2 is ILoanToken2, ERC20 {
         require(start.add(term).add(creditOracle.gracePeriod()) <= block.timestamp, "LoanToken2: Loan cannot be defaulted yet");
         status = Status.Defaulted;
 
-        emit Defaulted(_balance());
-    }
+        uint256 unpaidDebt = debt.sub(repaid());
+        debtToken = loanFactory.createDebtToken(pool, borrower, unpaidDebt);
 
-    /**
-     * @dev Liquidate the loan if it has defaulted
-     */
-    function liquidate() external override onlyDefaulted onlyLiquidator {
-        status = Status.Liquidated;
+        debtToken.approve(address(pool), unpaidDebt);
+        pool.addDebt(debtToken, unpaidDebt);
```

Comment:
Let's extract `pool` to a local variable.

Reply:
Looks like premature optimization. I'm pretty sure EVM optimizes can cache SLOADs from same location

Reply:
Ok, extracting pool would save us 190 gas out of 430k. Still not worth it